### PR TITLE
[Instrument] Add support for jsonData instrument in clearInstrument function

### LINKED ISSUE #7465: Clear Instrument fails with jsonData=true
When data is saved in an instrument with jsonData=true (ie it's saved in the flag table instead of an instrument table) `NDB_BVL_Instrument->clearInstrument()` throws an exception. ClearInstrument attempts to update the (non-existant) table.

In particular, this happens with the API handling a `PUT` request (since it tries to clear the old data before saving the new data).

The function should have an if statement to update the appropriate flag or instrument table based on the jsonData property.

### PR DESCRIPTION
**Pull request by ridz1208**
## Brief summary of changes

`clearInstrument` Function gives an error when the instrument has jsonData enabled. this fix adds support for clearInstrument by clear the data from the proper location

PS, make sure to ignore white spaces changes as most changes here are just indentation

#### Testing instructions (if applicable)

1. use the API to save an instrument

#### Link(s) to related issue(s)

* Resolves #7465

### CODE REVIEW

**Comment by GabrielPelletier** [comment]
Tested this on MCNRR and it all looks good.

File: php/libraries/NDB_BVL_Instrument.class.inc [hunk 1]
```diff
@@ -2300,29 +2300,33 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         $config   = NDB_Config::singleton();
         $db       = Database::singleton();
         $dbconfig = $config->getSetting('database');
-        $columns  = $db->pselect(
-            "SELECT COLUMN_NAME FROM information_schema.columns
+        if ($this->jsonData) {
+            $db->update('flag', ['Data'=>null], ['CommentID' => $this->commentID]);
```
Comment (driusan) [inline php/libraries/NDB_BVL_Instrument.class.inc:2304]: Shouldn't this update be unconditional? The `Data` column in the flag table exists regardless of whether jsonData is true or not, and the update when saving data in `_save` isn't conditional.

Comment (ridz1208) [reply php/libraries/NDB_BVL_Instrument.class.inc:2304]: You are correct. Ma bad

**Comment by ridz1208** [comment]
@driusan changes done

**Comment by GabrielPelletier** [comment]
Re-tested after the changes with MCNRR and it works alright.

**Review by driusan — APPROVED** [review]
I can settle for the downgrade from "good" to "alright" if it means it's ready to merge.

**Comment by ridz1208** [comment]
@driusan I think we should merge before @GabrielPelletier takes it from "alright" to "its not bad"

**Comment by GabrielPelletier** [comment]
I like a bit of variation; it makes for richer github Conversations, literature-wise

**Pull request merged**